src/plugins/rhbz.c

```diff
@@ -874,3 +890,164 @@ xmlrpc_value *rhbz_search_duphash(struct abrt_xmlrpc *ax,
+
+    /* Product info is stored in a struct as follows:
+     * {"products":({"id":i,"name":s,"description":s,"is_active":b,"default_milestone":s,"default_release":s,"has_unconfirmed":b,"classification":s,"components":({"id":i,"name":s,"description":s,"default_assigned_to":s,"default_qa_contact":s,"default_docs_contact":s,"default_cc":(s),"sort_key":i,"is_active":b,"sub_components":({})})})}
+     *
+     * where:
+     * () = array
+     * {} = struct
+     * b  = bool
+     * i  = int
+     * s  = string
+     * 
+     * The structs in the "sub_components" array are almost identical to the ones in the parent
+     * "components" array. The same is repeated recursively for subcomponents that themselves contain
+     * subcomponents.
+     * 
+     * http://xmlrpc-c.sourceforge.net/doc/libxmlrpc.html#formatstring
+     * 
+     * First, obtain a struct containing one item called "products" whose value is an array containing one struct
+     * containing information about the product we specified in the function arg. The "include_fields" arg
+     * specifies the struct as the following subset of the above:
+     * {"products":({"name":s,"components":({"name":s,"sub_components":({})})})}
+     * 
+     * The "components" struct doesn't have to be specified by itself because we specify its members.
+     * For some reason, the same doesn't work for "components.sub_components".
+     */
+
+    xmlrpc_value *products_array = abrt_xmlrpc_call(ax, "Product.get", "{s:(s),s:(ssss)}",
+                                                    "names", product,
+                                                    "include_fields",
+                                                    "name",
+                                                    "components.name",
+                                                    "components.sub_components",
+                                                    "components.sub_components.name");
+    if (env.fault_occurred)
+        abrt_xmlrpc_die(&env);
+
+    /* Extract the one product struct from the array.
+     * 'Actually, in the present implementation, you must specify this asterisk.
+     * That's because xmlrpc_decompose_value() is not capable of determining
+     * whether your format string extracts all the members of a structure,
+     * which is what no asterisk would imply. Some day, this will get fixed.'
+     * 
+     * http://xmlrpc-c.sourceforge.net/doc/libxmlrpc.html#decomposevalue
+     */
+    xmlrpc_value *product_struct = NULL;
+    xmlrpc_decompose_value(&env, products_array, "{s:(S),*}", "products", &product_struct);
+    if (env.fault_occurred)
+        abrt_xmlrpc_die(&env);
+
+    xmlrpc_value *comp_array = rhbz_get_member("components", product_struct);
+    if (env.fault_occurred)
+        abrt_xmlrpc_die(&env);
+
+    unsigned array_size = rhbz_array_size(comp_array);
+
+    const char *str_comp_name;
+    xmlrpc_value *comp_struct = NULL;
+
+    xmlrpc_value *comp_nameV = NULL;
```

Comment:
I think it would make more sense to limit the scope and have the cleanup be as close to allocation as possible. You don’t use this outside of the loop.

Reply:
Bonus points for not using the Hungarian notation.

Reply:
Yeah I know, I'm always running out of ideas in situations like this.

Reply:
There. I still don't like it, but purely as for Hungarianness it should be a little better.  
  
Cleanup moved too.